sr_utilities/src/sr_trajectory_command_publisher.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+* Copyright 2021 Shadow Robot Company Ltd.
+*
+* This program is free software: you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the Free
+* Software Foundation version 2 of the License.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <memory>
+#include <sr_utilities/sr_trajectory_command_publisher.hpp>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+void SrTrajectoryCommandPublisher::publish(trajectory_msgs::JointTrajectory joint_trajectory)
+{
+  std::unordered_map<std::shared_ptr<ros::Publisher>,
+    std::shared_ptr<trajectory_msgs::JointTrajectory>> publisher_to_msg;
+  for (int i = 0; i < joint_trajectory.joint_names.size(); i++)
+  {
+    std::string joint_name = joint_trajectory.joint_names[i];
```

Comment:
would be good to chunk it all up in some smaller methods. This code is really hard to read.